### PR TITLE
boards: arm: add olimex_lora_stm32wl_devkit

### PR DESCRIPTION
**Pull request by martinjaeger**
Add board definitions for [Olimex LoRa STM32WL DevKit](https://www.olimex.com/Products/IoT/LoRa/LoRa-STM32WL-DevKit/open-source-hardware).

### CODE REVIEW

File: boards/arm/olimex_lora_stm32wl_devkit/olimex_lora_stm32wl_devkit.dts [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+	pinctrl-0 = <&i2c1_scl_pa9 &i2c1_sda_pa10>;
+	pinctrl-names = "default";
+	status = "okay";
+	clock-frequency = <I2C_BITRATE_FAST>;
+
+
+	bme280: bme280@77 {
+		compatible = "bosch,bme280";
+		status = "okay";
+		label = "BME280";
```
Comment (galak) [inline boards/arm/olimex_lora_stm32wl_devkit/olimex_lora_stm32wl_devkit.dts:96]: remove label prop, not needed.

File: boards/arm/olimex_lora_stm32wl_devkit/olimex_lora_stm32wl_devkit.dts [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+		reg = <0x77>;
+	};
+
+	iis2mdc@1e {
+		compatible = "st,iis2mdc";
+		reg = <0x1e>;
+		drdy-gpios =  <&gpioa 8 GPIO_ACTIVE_HIGH>;
+		label = "IIS2MDC";
```
Comment (galak) [inline boards/arm/olimex_lora_stm32wl_devkit/olimex_lora_stm32wl_devkit.dts:104]: remove label prop, not needed.

**Review by galak — DISMISSED** [review]
minor updates in .dts requested

**Comment by martinjaeger** [comment]
The build failure is unrelated to this PR and will be fixed with #47557.

**Comment by martinjaeger** [comment]
Rebased to fix CI and add partitions to dts.

**Review by erwango — COMMENTED** [review]
One question, otherwise looks good.

File: boards/arm/olimex_lora_stm32wl_devkit/olimex_lora_stm32wl_devkit_defconfig [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+# PRELIMINARY: Don't enable MPU to prevent random MPU faults in Zephyr LoRa driver
+#CONFIG_ARM_MPU=y
+
+# PRELIMINARY: Don't enable HW stack protection (only available with MPU)
+#CONFIG_HW_STACK_PROTECTION=y
+
+# enable pin controller
+CONFIG_PINCTRL=y
+
+CONFIG_NVS=y
```
[the remark this answers is not in the document]
Comment (martinjaeger) [reply boards/arm/olimex_lora_stm32wl_devkit/olimex_lora_stm32wl_devkit_defconfig:26]: Yes, kind of, as LoRaWAN needs non-volatile storage to save the DevNonce. And as LoRaWAN is the primary use case for this board, I put it there.

Actually, thinking twice, we would also require `CONFIG_SETTINGS=y` and `CONFIG_LORAWAN_NVM_SETTINGS=y` for it to work, so I should maybe remove it again and leave it to the application / sample. What do you think?

[the remark this answers is not in the document]
Comment (erwango) [reply boards/arm/olimex_lora_stm32wl_devkit/olimex_lora_stm32wl_devkit_defconfig:26]: > Actually, thinking twice, we would also require CONFIG_SETTINGS=y and CONFIG_LORAWAN_NVM_SETTINGS=y for it to work, so I should maybe remove it again and leave it to the application / sample. What do you think?

An alternative is to move this to board's Kconfig.defconfig under  `if LORAWAN`.
But moving to sample is fine to me as well.

[the remark this answers is not in the document]
Comment (martinjaeger) [reply boards/arm/olimex_lora_stm32wl_devkit/olimex_lora_stm32wl_devkit_defconfig]: I've removed it and left configuration to the sample.

**Review by galak — DISMISSED** [review]

**Review by erwango — DISMISSED** [review]

**Comment by fabiobaltieri** [comment]
Hey @martinjaeger, since the device is a module + carrier board I would suggest splitting some of the dts nodes in a module include file, so that a user could include that in a custom board and not have to duplicate the radio frontend and clock config. This is pretty much the same setup as the Seeed E5 module so you may look into it as a reference (https://github.com/zephyrproject-rtos/zephyr/blob/main/dts/arm/seeed/lora-e5.dtsi). What do you think?

**Comment by martinjaeger** [comment]
I've seen that, but for me it was actually more confusing than helpful. The linked `.dtsi` basically only contains the LoRa chip part, but not e.g. pin assignments for UART, SPI and other peripherals which would also belong to the module. Also it is in a folder where I'd usually expect SOC devicetree definitions, but not board-related information.

I prefer to have the entire dev kit + module spec in one place, so that one just has to copy one folder and modify it for a custom board.

**Comment by fabiobaltieri** [comment]
> I've seen that, but for me it was actually more confusing than helpful. The linked `.dtsi` basically only contains the LoRa chip part, but not e.g. pin assignments for UART, SPI and other peripherals which would also belong to the module. Also it is in a folder where I'd usually expect SOC devicetree definitions, but not board-related information.
> 
> I prefer to have the entire dev kit + module spec in one place, so that one just has to copy one folder and modify it for a custom board.

Well but UART/SPI etc do not belong to the module, those pins so happen to be mapped to uart/i2c/spi devices in the specific carrier board but the WL55 has load of options for those pins that a carrier board designer may take. The radio frontend and clock (and the LED in this case I think), on the other hand, are part of the module, so there's no other possible configuration for it that a carrier board designer can take.

**Comment by martinjaeger** [comment]
Alright, moved the module dtsi part out of the board spec.

**Review by fabiobaltieri — APPROVED** [review]
Thanks!

**Review by erwango — APPROVED** [review]

**Pull request merged**